src/Features/CSharp/Portable/CodeFixes/ConditionalExpressionInStringInterpolation/CSharpAddParenthesisAroundConditionalExpressionInInterpolatedStringCodeFixProvider.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.ConditionalExpressionInStringInterpolation
+{
+
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.AddParenthesisAroundConditionalExpressionInInterpolatedString), Shared]
+    internal partial class CSharpAddParenthesisAroundConditionalExpressionInInterpolatedStringCodeFixProvider : CodeFixProvider
```

Comment:
Derive from SyntaxEditorBasedCodeFixProvider.  That way you can easily supply a fast and efficient 'FixAll' experience.

Reply:
FixAll is IMHO not useful for this kind of error (this fixes a syntax error and I don't see how there could be more then one instance of this error at a time). If there are other reasons I will change the base class (or if you just insists).

Reply:
You're def right that fix-all is less useful here.  where it could be valuable is for someone who changes a bunch of code, or copy  pastes a lot.  But if you don't want a fix-all, that's fine :)